migrations/committed/000002.sql

```diff
@@ -0,0 +1,190 @@
+      const attribute = attributes[0].find((a) => a.key === "code_id");
+      const code_id = attribute.value;
+      if (!attribute || !code_id) {
+        throw new Error("store code msg missing code ID from event");
+      }
+      const msg_id = `${tx_hash}-${id}`;
+      plv8.execute(
+        "INSERT INTO store_contract_messages (id, sender, permission, code_id, message_id, transaction_id, block_id) ($1, $2, $3, $4, $5, $6, $7)",
+        [msg_id, jsonMsg.sender, jsonMsg.permission, code_id, id, transaction_id, block_id]
+      );
+    }
+    for (const { id, json, transaction_id, block_id } of instantiateMsgResults) {
+      const jsonMsg = JSON.parse(json);
+      const [attributes, tx_hash] = getSelectResults(plv8.execute("select e.attributes, t.hash from messages m join transactions t on m.transaction_id = t.id join events e on t.id = e.transaction_id where m.id = $1;", id));
+      const attribute = attributes[0].find((a) => a.key === "code_id");
+      const code_id = attribute.value;
+      if (!attribute || !code_id) {
+        throw new Error("instantiate msg missing code ID from event");
+      }
+      const msg_id = `${tx_hash}-${id}`;
+      plv8.execute(
+        "INSERT INTO instantiate_contract_messages (id, sender, admin, code_id, label, payload, funds, message_id, transaction_id, block_id) ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)",
+        [msg_id, jsonMsg.sender, jsonMsg.admin, code_id, jsonMsg.label, jsonMsg.msg, jsonMsg.funds, id, transaction_id, block_id]
+      );
+    }
+    const instantiate_msgs = plv8.execute("select * from instantiate_contract_messages;");
+    for (const { id, payload, transaction_id } of instantiate_msgs) {
+      const [store_code_id, attributes] = getSelectResults(plv8.execute("select s.id, e.attributes from instantiate_contract_messages i join store_contract_messages s on i.code_id = s.code_id join messages m on i.message_id = m.id join transactions t on t.id = m.transaction_id join events e on t.id = e.transaction_id where i.id = $1;", [transaction_id]));
+      const payload_stripped = JSON.stringify(payload, null);
+      const attribute = attributes[0].find((a) => a.key === "_contract_address");
+      const contract_address = attribute.value;
+      if (!attribute || !contract_address) {
+        throw new Error("instantiate msg missing contract address from event");
+      }
+      plv8.execute("insert into contracts (id, interfaces, store_contract_message_id, instantiate_message_id) ($1, {$2}, $3, $4)", [contract_address, getJaccard(JSON.parse(payload_stripped)), id, store_code_id]);
+    }
+  }
+  return __toCommonJS(current_exports);
+})();
+
+return plv8ify.migrationAddInterfaceSupport()
+
+$plv8ify$ LANGUAGE plv8 IMMUTABLE STRICT;
+
+SELECT * from migrationAddInterfaceSupport();
```

Comment:
The name of the function as it's stored in Postgres ([000002.sql:8](https://github.com/fetchai/ledger-subquery/pull/174/files#diff-be76b025d211974f65a936059905d14123deddad3953db07871094d9c34a4540R8)) has a prefix (which I think we would have to remove manually, not sure it's worth it):
> ```sql
> CREATE OR REPLACE FUNCTION plv8ify_migrationAddInterfaceSupport() RETURNS JSONB AS $plv8ify$
> ```

```suggestion
SELECT * from plv8ify_migrationAddInterfaceSupport();
```

Reply:
I think it may add clarity to make a slight modification to the name; for example, "add contract interface detection" as opposed to "add interface support".